regression-test/suites/partition_p0/multi_partition/test_multi_partition.groovy

```diff
@@ -465,4 +465,25 @@ suite("test_multi_partition") {
     assertTrue(result2[1][1].startsWith("p_"))
     sql "drop table multi_par12"
 
-}
\ No newline at end of file
+
+    // create one table without datetime partition, but with date string
+    sql """
```

Comment:
```
mysql> set enable_fallback_to_original_planner=false;
Query OK, 0 rows affected (0.00 sec)

mysql> CREATE TABLE range_date_cast_to_datetime2 (id int, name string, pdate DATETIME) PARTITION BY RANGE(pdate)( FROM ("2023-04-16") TO ("2023-04-20") INTERVAL 1 DAY ) DISTRIBUTED BY HASH(id) BUCKETS 1 properties("replication_num" = "1");
Query OK, 0 rows affected (0.05 sec)
```

it can work well on nereids.